### PR TITLE
frontend/fix/#2319 : Fixes for subscription popups and banners.

### PR DESCRIPTION
**Pull request by prakharritik**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
Fixes for subscription popups and banners.
Made due component sticky.

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Frontend/ui-common/src/Components/PopUpModal/View.purs [hunk 1]
```diff
@@ -103,6 +103,7 @@ view push state =
                 , height WRAP_CONTENT
                 , margin $ MarginVertical 10 10
                 , color Color.black800
+                , gravity CENTER
```
Comment (MercyQueen) [inline Frontend/ui-common/src/Components/PopUpModal/View.purs:106]: @prakharritik can we make it configurable instead of adding gravity here directly.

File: Frontend/ui-common/src/Components/RideCompletedCard/Controller.purs [hunk 2]
```diff
@@ -142,7 +142,7 @@ config = {
   theme : DARK,
   isPrimaryButtonSticky : false,
   bannerConfig : Banner.config,
-  showBannerBelowQR : true
+  showBannerAboveQR : false
```
Comment (MercyQueen) [inline Frontend/ui-common/src/Components/RideCompletedCard/Controller.purs:145]: @prakharritik instead of maintaining a boolean variable to toggle, We can use a field to decide the order.

File: Frontend/ui-common/src/Engineering/Helpers/Commons.js [hunk 3]
```diff
@@ -493,6 +494,7 @@ function formatDates(date, format, language) {
     },
     'MMM': () => {
       const month = date.toLocaleDateString(language, { month: 'short' });
+      const month2 = date.toLocaleDateString(language, { month: 'long' });
```
Comment (MercyQueen) [inline Frontend/ui-common/src/Engineering/Helpers/Commons.js:497]: @prakharritik this variable is not being used anywhere. Is it required?

File: Frontend/ui-driver/src/Screens/HomeScreen/ComponentConfig.purs [hunk 4]
```diff
@@ -367,7 +368,9 @@ paymentPendingPopupConfig state =
     backgroundClickable = true,
     cornerRadius = (PTD.Corners 15.0 true true true true),
     coverImageConfig {
-      imageUrl = "ny_ic_payment_pending," <> getAssetStoreLink FunctionCall <> "ny_ic_payment_pending.png"
+      imageUrl = case popupType of
+                          GO_ONLINE_BLOCKER  -> "ny_ic_payment_pending," <> getAssetStoreLink FunctionCall <> "ny_ic_payment_pending.png"
+                          _ ->  "ny_ic_clear_dues_early," <> getAssetStoreLink FunctionCall <> "ny_ic_payment_pending.png"
```
Comment (MercyQueen) [inline Frontend/ui-driver/src/Screens/HomeScreen/ComponentConfig.purs:373]: @prakharritik is this asset added to asset store.

File: Frontend/ui-driver/src/Screens/SubscriptionScreen/Controller.purs [hunk 5]
```diff
@@ -121,8 +121,8 @@ eval BackPressed state =
   else if state.props.subView == ManagePlan then continue state{props { subView = MyPlan}}
   else if state.props.subView == DuesView then continue state{props { subView = MyPlan}}
   else if state.props.subView == DueDetails then 
-          if state.props.myPlanProps.multiTypeDues then continue state{props { subView = DuesView}}
-          else continue state{props { subView = MyPlan}}
+          if state.props.myPlanProps.multiTypeDues then continue state{props { subView = DuesView}, data{myPlanData{selectedDue = ""}}}
```
Comment (MercyQueen) [inline Frontend/ui-driver/src/Screens/SubscriptionScreen/Controller.purs:124]: use this condition only for changing the subview as that's the only parameter changing here.

File: Frontend/ui-driver/src/Strings/EN.js [hunk 6]
```diff
@@ -856,7 +856,7 @@ const englishStrings = {
   FAILURE : "FAILURE",
   PAYMENT_MODE : "Payment Mode",
   TXN_ID : "Order ID",
-  AMOUNT_PAID : "Amount Paid",
+  AMOUNT_PAID : "Amount",
```
Comment (MercyQueen) [inline Frontend/ui-driver/src/Strings/EN.js:859]: why is this change required, if we have replaced the constructor itself?

File: Frontend/ui-driver/src/Screens/PaymentHistoryScreen/Controller.purs [hunk 7]
```diff
@@ -111,22 +111,24 @@ getAutoPayInvoice (AutoPayInvoiceHistory autoPayInvoice) = {
...
+  ridesTakenDate : (convertUTCtoISC autoPayInvoice.rideTakenOn "Do MMM YYYY")
 }
 
 getManualPayInvoice :: ManualInvoiceHistory -> PaymentListItem
 getManualPayInvoice (ManualInvoiceHistory manualPayInvoice) = do
   let description' = case manualPayInvoice.feeType, (manualPayInvoice.rideDays - 1) of
                         AUTOPAY_REGISTRATION, 0 -> getString ONE_TIME_REGISTERATION
                         AUTOPAY_REGISTRATION, _ -> getString CLEARANCE_AND_REGISTERATION
-                        _, _ -> (getString RIDES_TAKEN_ON) <> " " <> show manualPayInvoice.rideDays <> " days"
+                        _, _ -> (getString RIDES_TAKEN_ON) <> " "
```
Comment (MercyQueen) [inline Frontend/ui-driver/src/Screens/PaymentHistoryScreen/Controller.purs:125]: don't we need the day after this string?

**Review by MercyQueen — CHANGES_REQUESTED** [review]

File: Frontend/ui-driver/src/Screens/HomeScreen/ComponentConfig.purs (hunk 4 above)
Comment (prakharritik) [reply Frontend/ui-driver/src/Screens/HomeScreen/ComponentConfig.purs:373]: Yes

File: Frontend/ui-common/src/Components/PopUpModal/View.purs (hunk 1 above)
Comment (prakharritik) [reply Frontend/ui-common/src/Components/PopUpModal/View.purs]: Done

File: Frontend/ui-common/src/Engineering/Helpers/Commons.js (hunk 3 above)
Comment (prakharritik) [reply Frontend/ui-common/src/Engineering/Helpers/Commons.js]: Fixed

File: Frontend/ui-driver/src/Screens/HomeScreen/ComponentConfig.purs (hunk 4 above)
Comment (MercyQueen) [reply Frontend/ui-driver/src/Screens/HomeScreen/ComponentConfig.purs:374]: @prakharritik the image name from native and remote doesn't match here.

File: Frontend/ui-driver/src/Screens/SubscriptionScreen/View.purs [hunk 8]
```diff
@@ -167,19 +168,38 @@ view push state =
...
-                  else linearLayout[][]
               ]
           ]
-        , BottomNavBar.view (push <<< BottomNavBarAction) (navData ScreenNames.SUBSCRIPTION_SCREEN)
+        , linearLayout
+          [ height MATCH_PARENT
+          , width MATCH_PARENT
+          , orientation VERTICAL
+          , background if state.props.myPlanProps.isDueViewExpanded && state.props.subView == MyPlan then Color.blackLessTrans else Color.transparent
+          , clickable if state.props.myPlanProps.isDueViewExpanded && state.props.subView == MyPlan then true else false
```
Comment (MercyQueen) [inline Frontend/ui-driver/src/Screens/SubscriptionScreen/View.purs:190]: assign the condition itself to clickable instead of using it as the condition

File: Frontend/ui-driver/src/Screens/SubscriptionScreen/View.purs [hunk 9]
```diff
@@ -562,275 +583,270 @@ headerView push state =
...
-                    pure unit
-                  )(const NoAction)
-  ][ 
-    linearLayout[
-        gravity CENTER
-      , padding $ Padding 16 16 16 8
+    [ height $ V 300 
+    , width MATCH_PARENT
+    , orientation VERTICAL
+    , padding $ PaddingVertical 24 (250)--layouts.height
```
Comment (MercyQueen) [inline Frontend/ui-driver/src/Screens/SubscriptionScreen/View.purs:594]: remove brackets and commented code

File: Frontend/ui-driver/src/Screens/SubscriptionScreen/View.purs [hunk 10]
```diff
@@ -562,275 +583,270 @@ headerView push state =
...
-        , tripList push state.data.myPlanData.dueItems (state.data.myPlanData.manualDueAmount /= 0.0) state.props.myPlanProps.isDuesExpanded true
-      ] 
-   ]
-   , if (state.data.myPlanData.autoPayStatus /= ACTIVE_AUTOPAY || state.data.myPlanData.manualDueAmount > 0.0) then PrimaryButton.view (push <<< ResumeAutoPay) (clearDueButtonConfig state) else dummyView -- 
-   , if (state.data.myPlanData.autoPayStatus /= ACTIVE_AUTOPAY && state.data.myPlanData.manualDueAmount > 0.0) then arrowButtonView push (getString ONE_TIME_SETTLEMENT) false OneTimeSettlement state.props.isSelectedLangTamil else dummyView -- 
-  ]
+          , tripList push state.data.myPlanData.dueItems (state.data.myPlanData.manualDueAmount /= 0.0) state.props.myPlanProps.isDuesExpanded true true
+        ] 
+    ]
+    , if (state.data.myPlanData.autoPayStatus /= ACTIVE_AUTOPAY || state.data.myPlanData.manualDueAmount > 0.0) then PrimaryButton.view (push <<< ResumeAutoPay) (clearDueButtonConfig state) else dummyView --
```
Comment (MercyQueen) [inline Frontend/ui-driver/src/Screens/SubscriptionScreen/View.purs:846]: remove `--` from this line

File: Frontend/ui-driver/src/Screens/SubscriptionScreen/View.purs [hunk 11]
```diff
@@ -167,19 +168,38 @@ view push state =
               height MATCH_PARENT
               , width MATCH_PARENT
               ][
-                joinPlanView push state (state.props.subView == JoinPlan)
-                , managePlanView push state (state.props.subView == ManagePlan)
-                , myPlanView push state (state.props.subView == MyPlan)
-                , autoPayDetailsView push state (state.props.subView == PlanDetails)
+                if (state.props.subView == JoinPlan) then joinPlanView push state (state.props.subView == JoinPlan) else dummyView
```
Comment (MercyQueen) [inline Frontend/ui-driver/src/Screens/SubscriptionScreen/View.purs:171]: can we do a case match for this instead

**Review by MercyQueen — CHANGES_REQUESTED** [review]

File: Frontend/ui-driver/src/Screens/PaymentHistoryScreen/View.purs [hunk 12]
```diff
@@ -153,72 +153,90 @@ paymentList push state =
...
+              , cornerRadius 8.0
+              , margin $ MarginVertical 6 6
+              , padding $ Padding 16 16 16 16
+              , onClick push $ const $ ListItemClick item
+              ][ linearLayout
+                  [ height WRAP_CONTENT
+                  , width MATCH_PARENT
+                  , gravity CENTER_VERTICAL
+                  , orientation HORIZONTAL
+                  , margin $ MarginVertical 0 6
```
Comment (MercyQueen) [inline Frontend/ui-driver/src/Screens/PaymentHistoryScreen/View.purs:178]: use marginBottom instead.

File: Frontend/ui-driver/src/Components/DueDetailsList/View.purs [hunk 13]
```diff
@@ -98,7 +98,7 @@ view push state =
                   , color Color.black800
                 ] <> FontStyle.h2 TypoGraphy
                 , imageView
-                  [ imageWithFallback if item.expanded == true then "ny_ic_chevron_up,https://assets.juspay.in/nammayatri/images/common/ny_ic_chevron_up.png" else "ny_ic_chevron_down,https://assets.juspay.in/nammayatri/images/common/ny_ic_chevron_down.png"
+                  [ imageWithFallback if item.expanded then "ny_ic_chevron_up,https://assets.juspay.in/nammayatri/images/common/ny_ic_chevron_up.png" else "ny_ic_chevron_down,https://assets.juspay.in/nammayatri/images/common/ny_ic_chevron_down.png"
```
Comment (Vignesh-772) [inline Frontend/ui-driver/src/Components/DueDetailsList/View.purs:101]: use `getCommonAssetStoreLink`

File: Frontend/ui-common/src/Components/RideCompletedCard/View.purs [hunk 14]
```diff
@@ -196,6 +196,16 @@ bottomCardView config push =
...
 
+getViewsByOrder :: forall w. Config -> RideCompletedElements -> (Action -> Effect Unit) -> PrestoDOM (Effect Unit) w
+getViewsByOrder config item push = 
+  case item of 
+    BANNER -> rideEndBannerView config.bannerConfig push
+    QR_VIEW -> driverUpiQrCodeView config push
+    NO_VPA_VIEW -> noVpaView config
+    BADGE_CARD -> badgeCardView config push
+    DRIVER_BOTTOM_VIEW -> driverBottomCardView config push
+    _ -> dummyTextView
```
Comment (Vignesh-772) [inline Frontend/ui-common/src/Components/RideCompletedCard/View.purs:207]: this default case is not required

File: Frontend/ui-driver/src/Screens/PaymentHistoryScreen/View.purs [hunk 15]
```diff
@@ -153,72 +153,90 @@ paymentList push state =
...
+                        , orientation HORIZONTAL
+                        , gravity CENTER_VERTICAL
+                        , cornerRadius 20.0
+                        , background Color.white900
+                        , padding $ Padding 8 5 8 5
+                        ][ imageView
+                            [ width $ V 12
+                            , height $ V 12
+                            , margin (MarginRight 4)
+                            , imageWithFallback "ny_ic_upi_logo,https://assets.juspay.in/beckn/nammayatri/driver/images/ny_ic_upi_logo.png"
```
Comment (Vignesh-772) [inline Frontend/ui-driver/src/Screens/PaymentHistoryScreen/View.purs:216]: here also use the function to base url

File: Frontend/ui-driver/src/Screens/SubscriptionScreen/View.purs [hunk 16]
```diff
@@ -362,7 +386,7 @@ plansBottomView push state =
           ][ textView $
               [ weight 1.0
               , height WRAP_CONTENT
-              , width $ V $ getWidthFromPercent 70
+              , width $ V $ JB.getWidthFromPercent 70
```
Comment (Vignesh-772) [inline Frontend/ui-driver/src/Screens/SubscriptionScreen/View.purs:389]: both weight and width will not work together. remove any one of them based on the requirement

File: Frontend/ui-driver/src/Screens/SubscriptionScreen/View.purs [hunk 17]
```diff
@@ -562,275 +588,270 @@ headerView push state =
...
+        , background Color.blue600
+        , cornerRadius 8.0
+        ][ linearLayout
+            [ height WRAP_CONTENT
+            , width MATCH_PARENT
+            , gravity CENTER_VERTICAL
+            ][ textView
+              [ text (getString CURRENT_DUES)
+              , textSize if state.props.isSelectedLangTamil then FontSize.a_10 else FontSize.a_12
+              , fontStyle $ FontStyle.medium LanguageStyle
```
Comment (Vignesh-772) [inline Frontend/ui-driver/src/Screens/SubscriptionScreen/View.purs:726]: append font props in all the text views

**Review by Vignesh-772 — CHANGES_REQUESTED** [review]

**Review by Vignesh-772 — APPROVED** [review]
